redaxo/src/core/tests/sql/sql_select_test.php

```diff
@@ -63,6 +63,48 @@ public function testGetRow()
         }
     }
 
+    public function testQueryType()
+    {
+        $allowedQueryTypes = [
+            'select',
+            'show',
+            'update',
+            'insert',
+            'delete',
+            'replace',
+            'create',
+            'call',
+            'optimize',
+        ];
+
+        foreach ($allowedQueryTypes as $queryType) {
```

Comment:
Besser wäre mit dataProvider zu arbeiten. Beispiel: https://github.com/redaxo/redaxo/blob/a1245ae74e9b1a4cde43a2232126ced8abb46beb/redaxo/src/core/tests/sql/sql_test.php#L105-L125

Das hat den Vorteil, dass man im Fehlerfall in der phpunit-Ausgabe sieht, welcher Datensatz genau fehlgeschlagen ist. Würde man hier bei der foreach-Variante nicht sehen.

Ich würde am Ende dann wahrscheinlich auch nur eine Testmethode nutzen, und ein dataprovider, und da dann sowohl Queries ohne Klammern, als auch welche mit, übergeben.

Reply:
Ja natürlich ist es besser, ich kenne DataProvider, habe nur nicht in dem Moment daran gedacht